File: distros/J/JavonetPerlSdk/lib/Javonet/Binaries/Python/javonet/core/handler/ArrayGetItemHandler.py
from javonet.core.handler.AbstractCommandHandler import AbstractCommandHandler


class ArrayGetItemHandler(AbstractCommandHandler):
    def __init__(self):
        self._required_parameters_count = 2

    def process(self, command):
        try:
            if len(command.payload) < self._required_parameters_count:
                raise Exception("ArrayGetItemHandler parameters mismatch!")

            array = command.payload[0]
            if isinstance(command.payload[1], list):
                indexes = command.payload[1]
            else:
                indexes = command.payload[1:]

            array_copy = array.copy()
            for i in indexes:
                array_copy = array_copy[i]
            return array_copy

        except Exception as e:
            exc_type, exc_value = type(e), e
            new_exc = exc_type(exc_value).with_traceback(e.__traceback__)
            raise new_exc from None

File: distros/J/JavonetPerlSdk/lib/Javonet/Binaries/Nodejs/core/handler/ArrayGetItemHandler.js
const AbstractHandler = require("./AbstractHandler")


class ArrayGetItemHandler extends AbstractHandler {
    requiredParametersCount = 2

    constructor() {
        super()
    }

    process(command) {
        try {
            if (command.payload.length < this.requiredParametersCount) {
                throw new Error("Array Get Item parameters mismatch")
            }
            let array = command.payload[0]
            let indexes
            if (Array.isArray(command.payload[1])) {
                indexes = command.payload[1]
            } else {
                indexes = command.payload.slice(1)
            }

            if (indexes.length === 1) {
                return array[indexes]
            } else {
                let array_copy = [...array]
                for (let i = 0; i < indexes.length; i++) {
                    array_copy = array_copy[indexes[i]]
                }
                return array_copy
            }
        } catch (error) {
            throw this.process_stack_trace(error, this.constructor.name)
        }
    }
}

module.exports = new ArrayGetItemHandler()